contracts/src/v0.8/feeds/DualAggregator.sol

```diff
@@ -680,18 +716,53 @@ contract DualAggregator is OCR2Abstract, OwnerIsCreator, AggregatorV2V3Interface
     bytes32[] calldata ss,
     bytes32 rawVs
   ) external override {
-    // TODO: update this function with implementation from the doc
-
     // NOTE: If the arguments to this function are changed, _requireExpectedMsgDataLength and/or
     // TRANSMIT_MSGDATA_CONSTANT_LENGTH_COMPONENT need to be changed accordingly
 
     uint256 initialGas = gasleft(); // This line must come first
+    Report memory report_;
 
-    HotVars memory hotVars = s_hotVars;
-
+    if (msg.sender == s_secondaryProxy) {
+      // Decode the report
+      report_ = _decodeReport(report);
+      (bool exist, uint32 roundId) = _reportExists(report_);
+      // In case the report exists, only copy the round id
+      if (exist) {
+        s_hotVars.latestSecondaryRoundId = roundId;
+        return;
+      } else {
+        // In case the report doesn't exist, lock the primary feed
+        primaryLocked = true;
```

Comment:
If it doesn't exist we need to make sure it's not an orphan report. Because if it is we want to ignore it.

Reply:
If it's an orphan report the execution will not succeed, the `_validateReport()` will revert it as a stale report:
```
if (epochAndRound < s_hotVars.latestEpochAndRound) {
    revert StaleReport();
  }
```